ouroboros-network/src/Ouroboros/Network/TxSubmission/Outbound.hs

```diff
@@ -80,10 +81,11 @@ txSubmissionOutbound
   => Tracer m (TraceTxSubmissionOutbound txid tx)
   -> NumTxIdsToAck  -- ^ Maximum number of unacknowledged txids allowed
   -> TxSubmissionMempoolReader txid tx idx m
+  -> (tx -> SizeInBytes) -- ^ get size of CBOR encoded transaction
   -> NodeToNodeVersion
   -> ControlMessageSTM m
   -> TxSubmissionClient txid tx m ()
-txSubmissionOutbound tracer maxUnacked TxSubmissionMempoolReader{..} _version controlMessageSTM =
+txSubmissionOutbound tracer maxUnacked TxSubmissionMempoolReader{..} _txSize _version controlMessageSTM =
```

Comment:
Does the outbound side need the callback, or can we use the sizes from the mempool?

Maybe we should change the mempool interface, to give us just `[txid]`s rather than `[(txid, SizeInBytes)]` - so we don't have multiple ways of getting that information, possibly in a non-compatible way.

Reply:
Essentially, the callback does provide the size of the transaction as it is stored in the mempool, except for the hash because it is not part of the transaction when crossing the network. I think your second remark is accurate.

Reply:
I've removed the callback from the outbound side. We can keep the interface as-is for now, and if/when the annotated codec gets integrated we can remove `SizeInBytes` from the mempool API.